src/can_service.py

```diff
@@ -118,6 +125,66 @@
 TEMP_ALARM_TOPIC = "alarms/temperature"
 LOAD_ALARM_TOPIC = "alarms/load"
 FUEL_ALARM_TOPIC = "alarms/fuel"
+lock = threading.Lock()
+
+
+def convert_to_tuple(protocol_data_entity):
```

Comment:
You should create a class for this instead of using tuple.

Reply:
Class already exists, not tuple is deleted and the class is used.